fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/runners/ExponentialDelayTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * ExponentialDelayTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.runners;
+
+import com.apple.foundationdb.async.AsyncUtil;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.RepeatedTest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ExponentialDelayTest {
```

Comment:
I think these tests are good, assuming they pass reliably, but I suppose an alternative would be to include assertions on the `currentDelay` state of the delay class, though that may be a little bit too fragile (that is to say, it relies on internal implementation details instead of the API output)

Reply:
Yeah, I could have asserted about the trend of `currentDelay`, but that I think I would still want tests like this, that are more functional.
I increased the repetitions for each of the 3 tests by a factor of 1000 (so 5,000; 20,000; 50,000) and they all passed (and it took 4 minutes)